### DISCUSSION TITLE
Keep window.showWarningMessage dialog open after clicking specific button

### DISCUSSION THREAD

**Discussion by hugomontero** [microsoft/vscode-discussions#73] [Extension Development QnA] [answered]
Hi everyone, I'm sharing here to look for some answers.
Sorry in advance for my bad english 😅 

The context is: I'm triggering a `window.showWarningMessage` toast dialog with three options:

<img width="535" alt="image" src="https://user-images.githubusercontent.com/6526979/180261516-6a6267ea-71af-4307-a42b-2c0b7ffac89d.png">

The thing I'm trying to do is:  When the user clicks on `learn more` option, then the message keeps open instead of being closed.

Is there any way to do that? I know the default behavior is when the user choose any option the message will close, also I couldn't see anything that allows me to change this behavior on the API.

At the end of the day I need to put a link or something that allows the user get more information about the message he's receiving, but keeps the message open in order to have the ability to choose "update" or "cancel". 

Thanks a lot in advance!

**Comment by bwateratmsft** [comment, marked as answer]
In the Azure extensions we do essentially exactly this: https://github.com/microsoft/vscode-azuretools/blob/main/utils/src/userInput/showWarningMessage.ts

The approach is to just reopen the exact same message if the user clicks "Learn More". It looks a little strange, but it gets the job done.

**Reply by hugomontero** [reply]
Thanks so much!! 😄 Yeah that was I imagined hehe. As you mentioned it looks a little strange hehe. I'll follow your advice and again thanks for your quick response!
